src/Our.Umbraco.UnVersion/Services/UnVersionService.cs

```diff
@@ -1,144 +1,207 @@
+                    continue;
+                }
+
+                // If we have a max days and the current version is older
+                if (configEntry.MaxDays > 0 && configEntry.MaxDays != int.MaxValue)
+                {
+                    var dateRemoveBefore = currentDateTime.AddDays(0 - configEntry.MaxDays);
+                    if (version.UpdateDate < dateRemoveBefore)
+                    {
+                        _logger.Debug<UnVersionService>("Remove version {versionId}, because version is updated {updateDate} and max days is {maxDays} (cutoff: {dateRemoveBefore})", version.VersionId, version.UpdateDate, configEntry.MaxDays, dateRemoveBefore);
+                        versionIdsToDelete.Add(version.VersionId);
+                    }
+                }
+
+            }
+
+            return versionIdsToDelete;
+
+        }
+
+        private List<int> GetNodeIdsFromXpath(string xpath)
+        {
+            using (var ctx = _context.EnsureUmbracoContext())
+            {
+                var nodes = ctx.UmbracoContext.Content.GetByXPath(xpath);
+
+                if (nodes == null)
+                    return new List<int>();
+
+                return nodes.Select(x => x.Id).ToList();
+            }
+        }
+
+        /// <summary>
+        /// Based on configEntries, deletes items from trash if too old or too many.
+        /// </summary>
+        public void CleanUpTrash()
+        {
+            _logger.Debug<CleanUpTrashTask>("CleanupTrashTask PerformRun start");
+
+            using (var scope = _scopeProvider.CreateScope())
+            {
+                using (var umbracoContextReference = _umbracoContextFactory.EnsureUmbracoContext())
+                {
+                    foreach (var cfg in _config.TrashConfigEntries)
+                    {
+                        var query = scope.SqlContext.Query<IContent>();
+
+                        if (cfg.DocTypeAlias != UnVersionConfig.AllDocumentTypesKey)
+                        {
+                            // The query does *not* like it when asked to match ContentType.Alias directly, so we get the ID instead
+                            var contentType = umbracoContextReference.UmbracoContext.Content.GetContentType(cfg.DocTypeAlias);
+
+                            if (contentType != null)
+                            {
+                                query = query.Where(x => x.ContentTypeId.Equals(contentType.Id));
+                            }
+                        }
+
+                        if (cfg.MaxDays < int.MaxValue)
+                        {
+                            // doesn't matter if config says -10 or 10, we always need a negative number
+                            var alwaysNegative = -Math.Abs(cfg.MaxDays); 
+
+                            var minDate = DateTime.Now.AddDays(alwaysNegative);
+
+                            query = query.Where(x => x.UpdateDate < minDate);
+                        }

```

Comment:
I think we need to handle MaxCount from config too, so if we take your config example, it should make sure that there is not more than 100 trashed newsPages trashed in the last 30 days.